src/lib/analyzeStream.ts

```diff
@@ -0,0 +1,41 @@
+import { Stream } from 'stream';
+import extractStreamMeta from './extractStreamMeta';
+import isStreamPNGAnimation from './isStreamPNGAnimation';
+import StreamSwitch from './stream-switch';
+
+interface ImageMetaData {
+  mimeType: string;
+  geometry: {
+    width: number,
+    height: number,
+  };
+  interlace: string;
+  channelDepth?: {
+    alpha?: number,
+  };
+  alpha?: string;
+  animatedGIF?: boolean;
+  animatedPNG?: boolean;
+}
+
+export default async (stream: Stream, requirements = []): Promise<ImageMetaData> => {
+  // XXX in case of face detection, analyze image for faces
+  const streamSwitch = new StreamSwitch(stream);
```

Comment:
Creating a new `StreamSwitch` here would potentially result in two full images being buffered in memory. How much of the stream does `extractStreamMeta` and `isStreamPNGAnimation` have to read? 

If it's not that much it would save quite some memory to "destroy" the StreamSwitch after the result has been obtained (after line 31). This means that the StreamSwitch will not buffer more than what is needed.

Reply:
Perhaps, as both methods consuming the streams are started at the same time its possible that no splitting of the stream is even necessary?

Reply:
could be so, would also rather see PassThrough here or preferably only a single command that detects everything.